server/scsynth/scsynth_main.cpp

```diff
@@ -360,11 +351,51 @@ int main(int argc, char* argv[]) {
 
     EventLoop::run([world]() { World_WaitForQuit(world, true); });
 
+    return 0;
+}
+
 #ifdef _WIN32
+
+int wmain(int argc, wchar_t** wargv) {
+    // initialize winsock
+    WSAData wsaData;
+    int nCode;
+    if ((nCode = WSAStartup(MAKEWORD(1, 1), &wsaData)) != 0) {
+        scprintf("WSAStartup() failed with error code %d.\n", nCode);
+        return 1;
+    }
+
+    // convert args to utf-8
+    std::vector<char*> argv;
+    for (int i = 0; i < argc; i++) {
+        auto argSize = WideCharToMultiByte(CP_UTF8, 0, wargv[i], -1, nullptr, 0, nullptr, nullptr);
+        argv.push_back(new char[argSize]);
+        WideCharToMultiByte(CP_UTF8, 0, wargv[i], -1, argv[i], argSize, nullptr, nullptr);
+    }
+
+    // set codepage to UTF-8
+    static UINT oldCodePage; // for remembering the old codepage when we switch to UTF-8
+    oldCodePage = GetConsoleOutputCP();
+    if (!SetConsoleOutputCP(65001))
+        scprintf("WARNING: could not set codepage to UTF-8\n");
+
+    // run main
+    int result = scsynth_main(argv.size(), argv.data());
+
     // clean up winsock
     WSACleanup();
+    // reset codepage from UTF-8
+    SetConsoleOutputCP(oldCodePage);
+    // clear vector with converted args
+    for (int i = 0; i < argv.size(); i++) {
+        delete[] argv[i];
```

Comment:
you can write this as a range-for, which is preferable in modern c++ when you don't need an index or iterator:

```cpp
for (auto* arg : argv)
    delete[] arg;
```

Reply:
fixed, hopefully!